kernel/mod_typing.ml

```diff
@@ -74,55 +74,46 @@ let rec check_with_def env struc (idl,(c,ctx)) mp equiv =
       (* In the spirit of subtyping.check_constant, we accept
          any implementations of parameters and opaques terms,
 	 as long as they have the right type *)
-      let c', univs, ctx' =
-        match cb.const_universes, ctx with
-        | Monomorphic_const _, None ->
-          let c',cst = match cb.const_body with
-            | Undef _ | OpaqueDef _ ->
-              let j = Typeops.infer env' c in
-              let typ = cb.const_type in
-              let cst' = Reduction.infer_conv_leq env' (Environ.universes env')
-                  j.uj_type typ in
-              j.uj_val, cst'
-            | Def cs ->
-              let c' = Mod_subst.force_constr cs in
-              c, Reduction.infer_conv env' (Environ.universes env') c c'
```

Comment:
Seems like this breaks the assumption that we only convert terms which have a common type.